transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/LegacyFlgDestination.java

```diff
@@ -46,22 +61,40 @@ public String getName() {
   @NonNull
   @Override
   public byte[] getExtras() {
-    return encodeString(apiKey);
+    return encodeAsExtras();
   }
 
   @NonNull
-  static byte[] encodeString(@NonNull String s) {
+  public static LegacyFlgDestination parseFromExtras(@NonNull byte[] a) {
     try {
-      return s.getBytes("UTF-8");
+      String encodedExtra = new String(a, "UTF-8");
+      if (!encodedExtra.startsWith(EXTRAS_VERSION_MARKER)) {
+        throw new IllegalStateException("Version marker missing from extras");
```

Comment:
I think `IllegalArgumentException` is more appropriate as it's not the state of the Destination that is invalid, but rather the input is invalid.